src/Calculator/WindowFrameService.cs

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using CalculatorApp.Common;
+using CalculatorApp.ViewModel.Common;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Windows.ApplicationModel.Core;
+using Windows.UI.Core;
+using Windows.UI.ViewManagement;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace CalculatorApp
+{
+    public sealed class WindowFrameService
+    {
+        public Page GetCurrentPage()
+        {
+            return (m_frame.Content as Page);
+        }
+
+        public void SetNewFrame(Windows.UI.Xaml.Controls.Frame frame)
+        {
+            Debug.Assert(frame.BackStackDepth == 0);
+            m_frame = frame;
+        }
+
+        // createdByUs means any window that we create.
+        // !createdByUs means the main window
+        internal static WindowFrameService CreateNewWindowFrameService(Frame viewFrame, bool createdByUs, WeakReference parent)
+        {
+            Debug.Assert(CoreWindow.GetForCurrentThread() != null);
+            var frameService = new WindowFrameService(viewFrame, parent);
+            frameService.InitializeFrameService(createdByUs);
+            return frameService;
+        }
+
+        public CoreDispatcher GetCoreDispatcher()
+        {
+            return m_coreDispatcher;
+        }
+
+        public int GetViewId()
+        {
+            return m_viewId;
+        }
+
+        public void RegisterOnWindowClosingHandler(Action onWindowClosingHandler)
+        {
+            m_onWindowClosingHandlers.Add(onWindowClosingHandler);
+        }
+
+        public Task HandleViewRelease()
+        {
+            var that = this;
```

Comment:
nit: there's no longer a reason to use `that`. `this` can be used in the lambda below.

Reply:
Good point - done